src/compile/common.ts

```diff
@@ -99,10 +99,19 @@ export function formatSignalRef(fieldDef: FieldDef<string>, specifiedFormat: str
   } else if (fieldDef.type === 'temporal') {
     const isUTCScale = isScaleFieldDef(fieldDef) && fieldDef['scale'] && fieldDef['scale'].type === ScaleType.UTC;
     return {
-      signal: timeFormatExpression(field(fieldDef, {expr}), fieldDef.timeUnit, specifiedFormat, config.text.shortTimeLabels, config.timeFormat, isUTCScale)
+      signal: timeFormatExpression(field(fieldDef, {expr}), fieldDef.timeUnit,
+        specifiedFormat, config.text.shortTimeLabels, config.timeFormat, isUTCScale)
     };
   } else {
-    return {signal: field(fieldDef, {expr})};
+    if (!specifiedFormat || specifiedFormat === 'number') {
```

Comment:
It's WIP. I haven't even finished the PR. Did I accidentally assign you to review it? I am sorry if I did :|